src/setup.js

```diff
@@ -118,27 +136,41 @@ function parseEntities(entities){
             entityDependence[name] = convertTypeToGQL(false, baseType);
         }
         let mapping = {};
+        let annotationMapping = {};
         if (properties != null && properties.length > 0){
             for (let property of properties){
                 let name = property['_attributes']['Name'];
                 let type = property['_attributes']['Type'];
                 mapping[name] = type;
+                let annotation = null;
+                if (property['Annotation']){
+                    annotation = property['Annotation']['_attributes']['String'];
+                }
+                annotationMapping[name] = annotation;
             }
         }
         if (navigationProperties != null && navigationProperties.length > 0){
             for (let navigationProperty of navigationProperties){
                 let name = navigationProperty['_attributes']['Name'];
                 let type = navigationProperty['_attributes']['Type'];
                 mapping[name] = type;
+                let annotation = null;
+                if (navigationProperty['Annotation']){
+                    annotation = navigationProperty['Annotation']['_attributes']['String'];
+                }
+                annotationMapping[name] = annotation;
             }
         }
         mapping['id'] = 'ID';
-        entityMapping[name] = mapping
+        entityAnnotationMapping[name] = annotationMapping;
+        entityMapping[name] = mapping;
     }
     return {"entityMapping": entityMapping,
-            "entityDependence": entityDependence};
+            "entityDependence": entityDependence,
```

Comment:
It looks like you made these changes twice and in general there's a lot of the same code between parseEntities and parseComplex. Is it worth it to move the common code to helpers or merge the functions?

Reply:
Yes, I think so. There are a few subtle differences between type parsing, but I think they're fairly minor.